core/market/decentralized/src/rest_api/error.rs

```diff
@@ -152,3 +152,21 @@ impl ResponseError for AgreementError {
         }
     }
 }
+
+impl ResponseError for WaitForApprovalError {
+    fn error_response(&self) -> HttpResponse {
+        let msg = ErrorMessage::new(self.to_string());
+        match self {
+            WaitForApprovalError::NotFound(_) => HttpResponse::NotFound().json(msg),
+            WaitForApprovalError::AgreementExpired(_) => HttpResponse::Gone().json(msg),
+            WaitForApprovalError::AgreementTerminated(_)
+            | WaitForApprovalError::AgreementNotConfirmed(_) => {
+                HttpResponse::BadRequest().json(msg)
```

Comment:
I believe we should return here `409 Conflict`, but it is not compatible with the spec. So spec needs to be updated